detox/src/utils/customConsoleLogger.js

```diff
@@ -0,0 +1,56 @@
+const util = require('util');
+const path = require('path');
+const callsites = require('./callsites');
+
+function callsitesToStack() {
+  return callsites().slice(2).reduce((result, callsite) => {
+    const filename = callsite.getFileName() ? path.relative(process.cwd(), callsite.getFileName()) : '';
+    const filePointer = filename ? `${filename}:${callsite.getLineNumber() || '?'}:${callsite.getColumnNumber() || '?'}` : '<unknown>';
+    const functionName = callsite.getFunctionName() || '<unknown>';
+    return result.concat(`\r    at ${functionName} (${filePointer})\n`)
+  }, '');
+}
+
+function getOrigin() {
+  const callsite = callsites()[2];
```

Comment:
What is this magic number 2, can we name it?

Reply:
it's because this is handled two-levels deep compared to the frame with the original user's call. I will name this indeed, you're right!